test/v1/controllers/requestParsers/RetrieveBalanceRequestParserSpec.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers
+
+import support.UnitSpec
+import uk.gov.hmrc.domain.Nino
+import v1.mocks.validators.MockRetrieveBalanceValidator
+import v1.models.errors.{ErrorWrapper, NinoFormatError}
+import v1.models.request.retrieveBalance.{RetrieveBalanceParsedRequest, RetrieveBalanceRawRequest}
+import v1.fixtures.RetrieveBalanceFixture._
+
+class RetrieveBalanceRequestParserSpec extends UnitSpec {
+
+  def inputData(nino: String): RetrieveBalanceRawRequest = RetrieveBalanceRawRequest(nino)
+
+  trait Test extends MockRetrieveBalanceValidator {
+    lazy val parser = new RetrieveBalanceRequestParser(mockValidator)
+  }
+
+  "parsing a retrieve Balance request" should {
+    "return a retrieve Balance request" when {
+      "valid data is provided" in new Test {
+
+        MockRetrieveBalanceValidator.validate(inputData(validNino))
+          .returns(Nil)
+
+        parser.parseRequest(inputData(validNino)) shouldBe
+          Right(RetrieveBalanceParsedRequest(Nino(validNino)))
+      }
+    }
+  }
+
+  "return an error" when {
+    "invalid data is provided" in new Test{
+      MockRetrieveBalanceValidator.validate(inputData(invalidNino))
+        .returns(List(NinoFormatError))
+
+      parser.parseRequest(inputData(invalidNino)) shouldBe
+        Left(ErrorWrapper(None, NinoFormatError, None))
+    }
+  }
+}
```

Comment:
inputData(validNino) and inputData(invalidNino) are both used multiple times so could be replaced by a val